### PR TITLE
Surround the statement if switch-case block contains declarations.

### LINKED ISSUE #1206: Shader compilation error on some windows hardware
In [this try push](https://treeherder.mozilla.org/#/jobs?repo=try&revision=db7e9f22bb8ab7da5820bdd071cedde86d894219) the windows 8 mochitest-bc failures happen because WR fails to startup. The log has this shader compilation error:

```
WebRender - OpenGL version new OpenGL ES 3.0 (ANGLE 2.1.0.dec065540d5f)
...
08:11:31     INFO - GECKO(4672) | Failed to link shader program: C:\fakepath(2369,1-2): error X3004: undeclared identifier 's3'
08:11:31     INFO - GECKO(4672) | Warning: D3D shader compilation failed with default flags. (vs_5_0)
08:11:31     INFO - GECKO(4672) |  Retrying with skip validation
08:11:31     INFO - GECKO(4672) | C:\fakepath(2369,1-2): error X3004: undeclared identifier 's3'
08:11:31     INFO - GECKO(4672) | Warning: D3D shader compilation failed with skip validation flags. (vs_5_0)
08:11:31     INFO - GECKO(4672) |  Retrying with skip optimization
08:11:31     INFO - GECKO(4672) | C:\fakepath(2369,1-2): error X3004: undeclared identifier 's3'
08:11:31     INFO - GECKO(4672) | Warning: D3D shader compilation failed with skip optimization flags. (vs_5_0)
08:11:31     INFO - GECKO(4672) | Failed to create D3D shaders.
```

Note however that this compilation failure happens pretty late in the log. AFAICT WebRender is initialized numerous times before that successfully. It might just be a shader that we're loading lazily and only gets loaded by particular tests.

Also note that the regular mochitest suite in the same try push must be running on different hardware because it seems to start WR just fine.

### PR DESCRIPTION
**Pull request by mephisto41**
This fix #1206 .

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1285)

### CODE REVIEW

**Comment by nical** [comment]
It would be great to add this to the [list of driver issues](https://github.com/servo/webrender/wiki/Driver-issues), even if this is not strictly coming from a driver. Or remind me to do it when I come back from vacation next week.

File: webrender/res/ps_border_edge.vs.glsl [hunk 1]
```diff
@@ -42,11 +42,15 @@ void write_color(vec4 color, float style, bool flip) {
 
     switch (int(style)) {
         case BORDER_STYLE_GROOVE:
+        {
```
Comment (kvark) [inline webrender/res/ps_border_edge.vs.glsl]: I don't see declarations in these switch cases. Are they also a problem?

Comment (mephisto41) [inline webrender/res/ps_border_edge.vs.glsl]: Forget to explain this case. See [1], ANGLE will translate ternary operator to ```type s; if(b) s = x; else s = y;```. And ```type s``` is a declaration.

[1]: https://github.com/google/angle/blob/d7b1ab5893953b97fe5dc1b3087a9b4b50ec45ee/src/compiler/translator/UnfoldShortCircuitToIf.cpp#L139

Comment (kvark) [inline webrender/res/ps_border_edge.vs.glsl]: omg, that's somewhat obscure
thanks for clarification!

**Comment by kvark** [comment]
I'll add it to the wiki.
Thanks for the fix!
@bors-servo r+

**Comment by kvark** [comment]
see https://github.com/servo/webrender/wiki/Driver-issues#1285---declarations-in-switch-cases

**Pull request merged**